persistence/gov.go

```diff
@@ -908,38 +105,51 @@ func (p PostgresContext) SetParam(paramName string, paramValue interface{}) erro
 	if err != nil {
 		return err
 	}
-	if _, err = tx.Exec(ctx, schema.NullifyParamsQuery(height)); err != nil {
-		return err
+
+	tableName := schema.ParamsTableName
+	if enabled != nil {
+		tableName = schema.FlagsTableName
 	}
-	if _, err = tx.Exec(ctx, schema.SetParam(paramName, paramValue, height)); err != nil {
+
+	if _, err = tx.Exec(ctx, schema.InsertParamOrFlag(tableName, paramName, height, paramValue, enabled)); err != nil {
 		return err
 	}
 	return tx.Commit(ctx)
 }
 
-func (p PostgresContext) GetIntParam(paramName string) (i int, err error) {
+func getParamOrFlag[T int | string | []byte](p PostgresContext, tableName, paramName string, height int64) (i T, enabled bool, err error) {
 	ctx, conn, err := p.GetCtxAndConnection()
 	if err != nil {
-		return 0, err
+		return i, enabled, err
 	}
-	err = conn.QueryRow(ctx, schema.GetParamQuery(paramName)).Scan(&i)
-	return
-}
 
-func (p PostgresContext) GetStringParam(paramName string) (s string, err error) {
-	ctx, conn, err := p.GetCtxAndConnection()
-	if err != nil {
-		return "", err
+	var stringVal string
+	row := conn.QueryRow(ctx, schema.GetParamOrFlagQuery(tableName, paramName, height))
+	if tableName == schema.ParamsTableName {
+		err = row.Scan(&stringVal)
+	} else {
+		err = row.Scan(&stringVal, &enabled)
 	}
-	err = conn.QueryRow(ctx, schema.GetParamQuery(paramName)).Scan(&s)
-	return
-}
-
-func (p PostgresContext) GetBytesParam(paramName string) (param []byte, err error) {
-	ctx, conn, err := p.GetCtxAndConnection()
 	if err != nil {
-		return nil, err
+		return
+	}
+	switch tp := any(i).(type) {
+	case int, int32, int64:
+		iConv, errr := strconv.Atoi(stringVal)
```

Comment:
Why not just use `err` here?

Reply:
replaced also below in line 143 `e` => `err`